### PR TITLE
GNIRS IFU implementation

### PR DESCRIPTION
**Pull request by rcooke-ast**
This is mostly a refactor of the `gemini_gnirs.py` file to accommodate the IFU modes in the future. No on-sky IFU (LR or HR) data currently exist, so a future PR will ingest data into the dev-suite and update this file as appropriate. Thanks go to Ray Sharples for contributing to this PR as well!!

I've also included a script that can be used to convert a wavelength solution file to something that can be used in the reid archive.

Running the dev-suite to make sure nothing broke in the process...

### CODE REVIEW

File: pypeit/core/wavecal/autoid.py
Comment (profxj) [inline pypeit/core/wavecal/autoid.py:2766]: thx for this;  always annoyed me :)

File: pypeit/scripts/arxiv_solution.py
Comment (profxj) [inline pypeit/scripts/arxiv_solution.py:1]: Nice!

**Review by profxj — APPROVED** [review]
Great stuff!

Post tests clearing before merging.
Thx!

File: pypeit/spectrographs/gemini_gnirs.py
Comment (debora-pe) [inline pypeit/spectrographs/gemini_gnirs.py]: The keys `comb_id` and `bkg_id` are added when we use the `-b` flag with `pypeit_setup`, we don't add them by default. I believe  `calib` is now always added.
Could you add here the key `filter`? It seems it's a useful information to have in the pypeit file

**Review by debora-pe — APPROVED** [review]
Thanks for this work @rcooke-ast . All looks good. I just wanted to let you know a few things.
On-sky low resolution IFU data does exist. It's the first light data taken almost a year ago. You can find all of them [here](https://archive.gemini.edu/searchform/GN-2022A-ENG-132/includeengineering/not_site_monitoring/NotFail/cols=CTOWBEQ), but I also selected a subset of those and put them in the Dev Suite Google Drive (in case you want to add a test in the dev suite - I also added a pypeit file for those). 
I did try to run the reduction and it run through fine except that I had to commented out [these lines](https://github.com/pypeit/PypeIt/blob/49e1897015794d9d741df435c88cd6069a11bd75/pypeit/find_objects.py#L1230-L1232) 
```
        if np.any(global_sky_sep[skymask] == 0):
            # Cannot continue without a sky model for all slits
            msgs.error("Global sky subtraction has failed for at least one slit.")
```

because for GNIRS there are some pixels outside of the slits that have global_sky == 0, so it seems that this condition is too restrictive.
I also tried to run `pypeit_coadd_datacube` and it run through fine, but I am not sure how to check the quality of the reduction.

That's everything. Thanks!

File: pypeit/spectrographs/gemini_gnirs.py
Comment (rcooke-ast) [reply pypeit/spectrographs/gemini_gnirs.py]: Good idea - updated.

**Comment by rcooke-ast** [comment]
> I just wanted to let you know a few things.

Thanks @debora-pe for the review and topping up the dev-suite!  I'll include this in the dev-suite PR...

**Comment by profxj** [comment]
Feel free to merge this @rcooke-ast if you are happy

**Comment by rcooke-ast** [comment]
Yep - I'll merge when dev-suite checks out!

**Comment by rcooke-ast** [comment]
> I did try to run the reduction and it run through fine except that I had to commented out [these lines](https://github.com/pypeit/PypeIt/blob/49e1897015794d9d741df435c88cd6069a11bd75/pypeit/find_objects.py#L1230-L1232)

Hi @debora-pe - Thanks again for providing the test data in the dev-suite. I've fixed this error, but upon checking all of the QA, I noticed a slight issue in the illumflat determination. I think this is fine for now, but I'll look into this.

I also need to check the datacubes looks good  :-)

**Comment by debora-pe** [comment]
> > I did try to run the reduction and it run through fine except that I had to commented out [these lines](https://github.com/pypeit/PypeIt/blob/49e1897015794d9d741df435c88cd6069a11bd75/pypeit/find_objects.py#L1230-L1232)
> 
> Hi @debora-pe - Thanks again for providing the test data in the dev-suite. I've fixed this error, but upon checking all of the QA, I noticed a slight issue in the illumflat determination. I think this is fine for now, but I'll look into this.
> 
> I also need to check the datacubes looks good :-)

Sounds good! Thanks @rcooke-ast !

**Comment by rcooke-ast** [comment]
Tests passing...
image

**Pull request merged**
